datapackage/resource.py

```diff
@@ -268,8 +268,14 @@ def iter(self):
             inline_data = self._parse_inline_data()
             result = iter(inline_data)
         elif data_path_or_url:
+            dialect = self.metadata.get('dialect', {})
+            parser_options = {}
+            if 'delimiter' in dialect:
+                parser_options['delimiter'] = dialect['delimiter']
+            if 'lineTerminator' in dialect:
+                parser_options['lineterminator'] = dialect['lineTerminator']
             try:
-                table = tabulator.topen(data_path_or_url, with_headers=True)
+                table = tabulator.topen(data_path_or_url, with_headers=True, parser_options=parser_options)
```

Comment:
Good practice here in topen call will be to add explicit `parser_class=tabulator.parsers.CSV` because the options is definitely for csv parser (also we don't want to ensure csv parser will be used)